frontend/src/component/layout/MainLayout/NavigationSidebar/ConfigurationAccordion.tsx

```diff
@@ -0,0 +1,77 @@
+import { type FC, useEffect, useState } from 'react';
+import { MenuListAccordion } from './ListItems.tsx';
+import { useExpanded } from './useExpanded.ts';
+import type { NavigationMode } from './NavigationMode.tsx';
+import { IconRenderer } from './IconRenderer.tsx';
+import { ConfigurationNavigationList } from './ConfigurationNavigationList.tsx';
+import { useRoutes } from './useRoutes.ts';
+
+type ConfigurationAccordionProps = {
+    mode: NavigationMode;
+    setMode;
+    activeItem?: string;
+    onClick?: () => void;
+};
+
+export const ConfigurationAccordion: FC<ConfigurationAccordionProps> = ({
+    mode,
+    setMode,
+    activeItem,
+    onClick,
+}) => {
+    const [expanded, changeExpanded] = useExpanded<'configure'>();
+    const [tempExpanded, setTempExpanded] = useState(false);
+    const { routes } = useRoutes();
+    const subRoutes = routes.mainNavRoutes;
+    const hasActiveItem = Boolean(
+        activeItem && subRoutes.some((route) => route.path === activeItem),
+    );
+
+    useEffect(() => {
+        if (mode === 'mini') {
+            setTempExpanded(false);
+        }
+    }, [mode]);
+
+    const onExpandChange = () => {
+        changeExpanded('configure', !expanded.includes('configure'));
+
+        if (tempExpanded) {
+            setTempExpanded(false);
+            setMode('mini');
+        }
+        if (mode === 'mini') {
+            setTempExpanded(true);
+            setMode('full');
+        }
+    };
+
+    const onItemClick = () => {
+        if (tempExpanded) {
+            setTempExpanded(false);
+            setMode('mini');
+        }
+        onClick?.();
+    };
```

Comment:
if "full", it will just expand/collapse as normal. If 'mini', it will open "full" mode until "configure" is collapsed back.